src/js/providers/flash.js

```diff
@@ -222,7 +222,6 @@ define([
                         this.sendEvent(events.JWPLAYER_MEDIA_ERROR, {
                             message: 'Error loading media: File could not be played'
                         });
-                        this.setState(states.IDLE);
```

Comment:
Thanks for the explanation, it makes a lot of sense to set the state in the controller. When this is mergable ping me and I'll put it in, or feel free to merge it yourself.